editor/components/block-edit/index.js

```diff
@@ -52,18 +46,14 @@ export class BlockEdit extends Component {
 		} );
 	}
 
-	static getDerivedStateFromProps( { name, isSelected }, prevState ) {
-		if (
-			name === prevState.name &&
-			isSelected === prevState.isSelected
-		) {
-			return null;
-		}
+	static getDerivedStateFromProps( props ) {
+		const { id, name, isSelected, renderBlockMenu } = props;
 
 		return {
-			...prevState,
```

Comment:
As part of #6419 (ead8dc5b46f3c801122d925ee9a4019cb7a40ef8), I'm guessing this was previously trying to avoid clobbering the `focusedElement` and `setFocusedElement` assigned in `constructor`. From my testing, this doesn't happen: The `return` value from `getDerivedStateFromProps` is applied as patch (same as `setState`):

https://codepen.io/aduth/pen/YvGyag

cc @gziolo

Reply:
I will test it tomorrow. I think it was introduced before and was trying to prevent state updates when props didn’t change by returning null. Maybe your proposal works the same, it would be awesome 😃

Reply:
I found my commit which I used to test `BlockEdit` updates: https://github.com/WordPress/gutenberg/commit/909844f4966263422177d93e8c947ab879a078c9. Great refactoring, everything works as expected. I didn't discover any regression and the code is much cleaner 💯